pennylane/interfaces/tf.py

```diff
@@ -138,34 +138,82 @@ def _execute(self, params, **input_kwargs):
         res = self.execute_device(args, input_kwargs["device"])
         self.set_parameters(all_params, trainable_only=False)
 
-        def grad(grad_output, **tfkwargs):
-            variables = tfkwargs.get("variables", None)
+        saved_grad_matrices = {}
+
+        def _evaluate_grad_matrix(grad_matrix_fn):
+            """Convenience function for generating gradient matrices
+            for the given parameter values.
+
+            This function serves two purposes:
+
+            * Avoids duplicating logic surrounding parameter unwrapping/wrapping
+
+            * Takes advantage of closure, to cache computed gradient matrices via
+              saved_grad_matrices, to avoid gradient matrices being
+              computed multiple redundant times.
+
+              This is particularly useful when differentiating vector-valued QNodes.
+              Because TensorFlow requests the vector-GradMatrix product,
+              and *not* the full GradMatrix, differentiating vector-valued

```

Comment:
Why is it written `GradMatrix` in camel case? I think it looks like a C++ class name like this. 🤔

Reply:
Updated to remove the camel case